CalledProcessError in test_4_cwl_rnaseq

**Version info**
 - bcbio version (bcbio_nextgen.py --version): 1.2.0 (a0db761)
 - OS name and version: Ubuntu Bionic

**To Reproduce**
 - Exact bcbio command you have used: `py.test -p no:cacheprovider -p no:stepwise tests/bcbio_vm -v -s -m cwl_docker_rnaseq`
 - Your sample configuration file: N/A

**Observed behavior**
```
=================================== FAILURES ===================================
__________________________ TestCWL.test_4_cwl_rnaseq ___________________________

self = <test_docker.TestCWL object at 0x7f34f77cfa50>, install_test_files = None
data_dir = '/home/travis/build/bcbio/bcbio-nextgen/tests/data/automated'

    @pytest.mark.cwl
    @pytest.mark.cwl_docker_rnaseq
    def test_4_cwl_rnaseq(self, install_test_files, data_dir):
        """CWL: run a RNA-seq workflow using Docker.
        """
        with install_cwl_test_files(data_dir) as workdir:
            with utils.chdir(os.path.join(workdir, "rnaseq")):
                cl = ["bash", "./run_generate_cwl.sh"]
                if os.path.exists("cromwell_work"):
                    shutil.rmtree("cromwell_work")
                cl = ["bcbio_vm.py", "cwlrun", "cromwell", "rnaseq-workflow"]
>               subprocess.check_call(cl)

tests/bcbio_vm/test_docker.py:99: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

popenargs = (['bcbio_vm.py', 'cwlrun', 'cromwell', 'rnaseq-workflow'],)
kwargs = {}, retcode = 1
cmd = ['bcbio_vm.py', 'cwlrun', 'cromwell', 'rnaseq-workflow']

    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.
    
        The arguments are the same as for the call function.  Example:
    
        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['bcbio_vm.py', 'cwlrun', 'cromwell', 'rnaseq-workflow']' returned non-zero exit status 1.

../../../install/bcbio-vm/anaconda/lib/python3.7/subprocess.py:363: CalledProcessError
```

**Expected behavior**
No error

**Log files**
https://travis-ci.org/github/bcbio/bcbio-nextgen/jobs/666050606

**Additional context**
In Vagrant:
```
cd tests/data/test_bcbio_cwl-master/rnaseq
bcbio_vm.py cwlrun cromwell rnaseq-workflow
```
Relevant output:
```
[2020-03-25 11:34:00,92] [info] WorkflowManagerActor Workflow 638ba748-3d9d-4e32-a5d5-98233d140eac failed (during ExecutingWorkflowState): cromwell.backend.standard.StandardAsyncExecutionActor$$anon$2: Failed to evaluate job outputs:
Bad output 'prepare_sample.prep_rec': No output binding is defined. Are you expecting the output to be inferred from a cwl.output.json file ? If so please make sure the file was effectively created.

Workflow 638ba748-3d9d-4e32-a5d5-98233d140eac transitioned to state Failed
Failed bcbio Cromwell run
-------------------------
Failure in step: prepare_sample.0
  bcbio log file     : cromwell_work/cromwell-executions/main-rnaseq.cwl/638ba748-3d9d-4e32-a5d5-98233d140eac/call-prepare_sample/shard-0/execution/log/bcbio-nextgen-debug.log
  bcbio commands file: cromwell_work/cromwell-executions/main-rnaseq.cwl/638ba748-3d9d-4e32-a5d5-98233d140eac/call-prepare_sample/shard-0/execution/log/bcbio-nextgen-commands.log
  Cromwell directory : cromwell_work/cromwell-executions/main-rnaseq.cwl/638ba748-3d9d-4e32-a5d5-98233d140eac/call-prepare_sample/shard-0
```

--- a/tests/bcbio_vm/test_docker.py
+++ b/tests/bcbio_vm/test_docker.py
@@ -50,7 +50,7 @@ def test_docker_ipython(install_test_files, data_dir):
         subprocess.check_call(cl)
 
 
-class TestCWL():
+class TestCWL:
     """ Run simple CWL workflows.
 
     Requires https://github.com/bcbio/bcbio-nextgen-vm
@@ -58,70 +58,50 @@ class TestCWL():
     @pytest.mark.cwl
     @pytest.mark.cwl_docker
     @pytest.mark.cwl_docker_somatic
-    def test_2_cwl_docker_somatic(self, data_dir):
-        """CWL: run a somatic workflow using Docker.
-        """
-        with install_cwl_test_files(data_dir) as workdir:
+    def test_cwl_docker_somatic_workflow(self):
+        with install_cwl_test_files() as workdir:
             with utils.chdir(os.path.join(workdir, "somatic")):
-                cl = ["bash", "./run_generate_cwl.sh"]
-                subprocess.check_call(cl)
+                subprocess.check_call(["bash", "./run_generate_cwl.sh"])
                 if os.path.exists("cromwell_work"):
                     shutil.rmtree("cromwell_work")
-                cl = ["bcbio_vm.py", "cwlrun", "cromwell", "somatic-workflow"]
-                subprocess.check_call(cl)
+                subprocess.check_call(["bcbio_vm.py", "cwlrun", "cromwell", "somatic-workflow"])
 
     @pytest.mark.cwl
     @pytest.mark.cwl_docker
     @pytest.mark.cwl_docker_joint
-    def test_3_cwl_docker_joint(self, data_dir):
-        """CWL: run a gVCF based joint-calling workflow using Docker.
-        """
-        with install_cwl_test_files(data_dir) as workdir:
+    def test_cwl_docker_joint_calling_workflow(self):
+        with install_cwl_test_files() as workdir:
             with utils.chdir(os.path.join(workdir, "gvcf_joint")):
-                cl = ["bash", "./run_generate_cwl.sh"]
-                subprocess.check_call(cl)
+                subprocess.check_call(["bash", "./run_generate_cwl.sh"])
                 if os.path.exists("cromwell_work"):
                     shutil.rmtree("cromwell_work")
-                cl = ["bash", "./run_cromwell.sh"]
-                subprocess.check_call(cl)
+                subprocess.check_call(["bash", "./run_cromwell.sh"])
 
     @pytest.mark.cwl
     @pytest.mark.cwl_docker_rnaseq
-    def test_4_cwl_rnaseq(self, install_test_files, data_dir):
-        """CWL: run a RNA-seq workflow using Docker.
-        """
-        with install_cwl_test_files(data_dir) as workdir:
-            with utils.chdir(os.path.join(workdir, "rnaseq")):
-                cl = ["bash", "./run_generate_cwl.sh"]
+    def test_cwl_rnaseq(self, install_test_files):
+        with install_cwl_test_files() as work_dir:
+            with utils.chdir(os.path.join(work_dir, "rnaseq")):
                 if os.path.exists("cromwell_work"):
                     shutil.rmtree("cromwell_work")
-                cl = ["bcbio_vm.py", "cwlrun", "cromwell", "rnaseq-workflow"]
-                subprocess.check_call(cl)
+                subprocess.check_call(["bcbio_vm.py", "cwlrun", "cromwell", "rnaseq-workflow"])
 
     @pytest.mark.cwl
     @pytest.mark.cwl_local
     @pytest.mark.install_required
-    def test_1_cwl_local(self, install_test_files, data_dir):
-        """CWL: prepare somatic workflow and run on local installation.
-        """
-        with install_cwl_test_files(data_dir) as workdir:
+    def test_cwl_local_somatic_workflow(self, install_test_files):
+        with install_cwl_test_files() as workdir:
             with utils.chdir(os.path.join(workdir, "somatic")):
-                cl = ["bash", "./run_generate_cwl.sh"]
-                subprocess.check_call(cl)
+                subprocess.check_call(["bash", "./run_generate_cwl.sh"])
                 if os.path.exists("cwltool_work"):
                     shutil.rmtree("cwltool_work")
-                cl = ["bash", "./run_cwltool.sh"]
-                subprocess.check_call(cl)
+                subprocess.check_call(["bash", "./run_cwltool.sh"])
 
     @pytest.mark.cwl
     @pytest.mark.cwl_arvados
-    def test_5_cwl_arvados(self, install_test_files, data_dir):
-        """CWL: prepare workflow for running on Arvados.
-
-        Requires ARVADOS_API_HOST and ARVADOS_API_TOKEN set
-        """
+    def test_cwl_arvados_workflow(self, install_test_files):
+        """Requires ARVADOS_API_HOST and ARVADOS_API_TOKEN set"""
         if os.environ.get("ARVADOS_API_HOST") and os.environ.get("ARVADOS_API_TOKEN"):
-            with install_cwl_test_files(data_dir) as workdir:
+            with install_cwl_test_files() as workdir:
                 with utils.chdir(os.path.join(workdir, "arvados")):
-                    cl = ["bash", "./run_generate_cwl.sh"]
-                    subprocess.check_call(cl)
+                    subprocess.check_call(["bash", "./run_generate_cwl.sh"])

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -78,11 +78,13 @@ def get_post_process_yaml(data_dir, workdir):
             yaml.dump(config, out_handle)
     return test_system
 
+
 @contextlib.contextmanager
-def install_cwl_test_files(data_dir):
+def install_cwl_test_files():
     orig_dir = os.getcwd()
     url = "https://github.com/bcbio/test_bcbio_cwl/archive/master.tar.gz"
-    dirname = os.path.normpath(os.path.join(data_dir, os.pardir, "test_bcbio_cwl-master"))
+    test_dir = os.getenv("BCBIO_TEST_DIR", test_data_dir())
+    dirname = os.path.join(test_dir, "test_bcbio_cwl-master")
     if os.path.exists(dirname):
         # check for updated commits if the directory exists
         ctime = os.path.getctime(os.path.join(dirname, "README.md"))
@@ -93,6 +95,7 @@ def install_cwl_test_files(data_dir):
     try:
         if not os.path.exists(dirname):
             print("Downloading CWL test directory: %s" % url)
+            os.makedirs(dirname)
             os.chdir(os.path.dirname(dirname))
             r = requests.get(url)
             tf = tarfile.open(fileobj=io.BytesIO(r.content), mode='r|gz')
@@ -101,6 +104,7 @@ def install_cwl_test_files(data_dir):
         yield dirname
     finally:
         os.chdir(orig_dir)
+
 
 def _get_bcbio_system(workdir, data_dir):
     system = _get_bcbiovm_config(data_dir)

